src/common/error/src/lib.rs

```diff
@@ -18,9 +18,22 @@ pub mod ext;
 pub mod mock;
 pub mod status_code;
 
+use http::{HeaderMap, HeaderValue};
 pub use snafu;
 
 // HACK - these headers are here for shared in gRPC services. For common HTTP headers,
 // please define in `src/servers/src/http/header.rs`.
 pub const GREPTIME_DB_HEADER_ERROR_CODE: &str = "x-greptime-err-code";
 pub const GREPTIME_DB_HEADER_ERROR_MSG: &str = "x-greptime-err-msg";
+
+/// Create a http header map from error code and message.
+/// using `GREPTIME_DB_HEADER_ERROR_CODE` and `GREPTIME_DB_HEADER_ERROR_MSG` as keys.
+pub fn from_err_code_msg_to_header(
+    code: impl Into<HeaderValue>,
+    msg: impl Into<HeaderValue>,
+) -> HeaderMap {
```

Comment:
Does `TypedHeader` suit this case better? https://docs.rs/axum/0.6.19/axum/struct.TypedHeader.html

Reply:
> Does `TypedHeader` suit this case better? https://docs.rs/axum/0.6.19/axum/struct.TypedHeader.html

I can't find it in newer version of axum, so no?:
https://docs.rs/axum/0.7.9/axum/?search=TypedHeader